### ISSUE TITLE
DL3044 false positive with an escaped dollar ($)

### ISSUE THREAD

**Issue by dosyfier** [hadolint/hadolint#1062]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

The following declaration in a Dockerfile should not throw any Hadolint warning:

```Dockerfile
ENV SOME_VARIABLE=rw \
    SOME_OTHER_VARIABLE="\${SOME_VARIABLE}_additional_content"
```

Note: The rationale behind such construct is to be able to override SOME_VARIABLE when running a container, and expand SOME_OTHER_VARIABLE via the entrypoint, with the new value of SOME_VARIABLE (or the default one if it's not provided to `docker run [...]`).

### Actual behavior

It does throw the following error:

```
DL3044 error: Do not refer to an environment variable within the same `ENV` statement where it is defined.
```

### Steps to reproduce the behavior

* `hadolint --version`: Haskell Dockerfile Linter 2.12.0
* Run `hadolint <FILE>` with a Dockerfile containing the above `ENV` declaration.
